crates/sui-types/src/execution_status.rs

```diff
@@ -305,6 +305,16 @@ pub enum CommandArgumentError {
         multiple arguments."
     )]
     InvalidArgumentArity,
+    #[error(
+        "Object passed to TransferObject does not have public transfer, i.e. the `store` \
+        ability"
+    )]
+    InvalidTransferObject,
+    #[error(
+        "When MakeMoveVec is used without a type specified, all arguments must be the same \
+        object type. In this case, the first argument was not an object."
```

Comment:
Nit: maybe consider rewording this error message to be a bit more in-line with the tone of the other error message descriptions? e.g.,
```suggestion
        "First argument to MakeMoveVec is not an object. If no type is specified for MakeMoveVec all arguments must be the same object type."
```